NumberValidator/Helpers/ExtensionMethods.cs

```diff
@@ -22,5 +22,17 @@ public static bool IsDigits(this string input)
 
             return digitsRegularExpression.Match(input).Success;
         }
+
+        public static string RemoveSpace(this  string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+
+            var cleanedInput = Regex.Replace(input, @"[\s-]", "");
+
+            return cleanedInput.ToUpper().Trim();
```

Comment:
there is a coding principle which is called `Single Responsibility Principle`. In a method which is saying that it `RemoveSpace`, you are also converting it ToUpper and Trimming which is not good. leave those aspects of the logic to other places in the code. Make this method handle only `RemoveSpace` part.

Reply:
sure , i will change it.

Reply:
changes done